extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/datatransferproject/transfer/smugmug/photos/SmugMugPhotosImporter.java

```diff
@@ -106,15 +106,15 @@ void importSingleAlbum(UUID jobId, PhotoAlbum inputAlbum, SmugMugInterface smugM
       jobStore.create(jobId, createCacheKey(), tempPhotosData);
     }
     tempPhotosData.addAlbumId(inputAlbum.getId(), response.getUri());
+    jobStore.update(jobId, createCacheKey(), tempPhotosData);
   }
 
   @VisibleForTesting
   void importSinglePhoto(UUID jobId, PhotoModel inputPhoto, SmugMugInterface smugMugInterface)
       throws IOException {
     // Find album to upload photo to
-    String newAlbumUri =
-        jobStore.findData(jobId, createCacheKey(), TempPhotosData.class)
-            .lookupNewAlbumId(inputPhoto.getAlbumId());
+    TempPhotosData tempPhotosData = jobStore.findData(jobId, createCacheKey(), TempPhotosData.class);
+    String newAlbumUri = tempPhotosData.lookupNewAlbumId(inputPhoto.getAlbumId());
```

Comment:
Let's log if it's empty

Reply:
Added logging to RetryingCallable, should log every error that it catches.